pkg/querypb/query_response.proto

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+// Provenance-includes-location: https://github.com/prometheus/prometheus/blob/main/prompb/types.proto
+// Provenance-includes-license: Apache-2.0
+// Provenance-includes-copyright: Prometheus Team
+
+syntax = "proto3";
+
+package querypb;
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+
+message QueryResponse {
+  // These values correspond to the possible status values defined in https://github.com/prometheus/prometheus/blob/main/web/api/v1/api.go.
+  enum Status {
```

Comment:
This enum (`Status`) and `ErrorType` below need to stay in sync with the possible values defined in Prometheus.

Alternatively, if we don't want to maintain these here, we could send strings for `status` and `error_type`, but this would increase payload size and allocations (we'd allocate an instance of `"success"` for every query payload decoded).